cnd/src/swap_protocols/asset.rs

```diff
@@ -21,22 +21,40 @@ pub trait Asset:
     + FromHttpAsset
     + Into<AssetKind>
 {
-    fn equal_or_greater_value(&self, other: &Self) -> bool;
+    fn compare_to(&self, other: &Self) -> i8;
 }
 
 impl Asset for BitcoinQuantity {
-    fn equal_or_greater_value(&self, other: &BitcoinQuantity) -> bool {
-        self >= other
+    fn compare_to(&self, other: &BitcoinQuantity) -> i8 {
+        if self < other {
+            return -1;
+        }
+        if self > other {
+            return 1;
+        }
+        return 0;
     }
 }
 impl Asset for EtherQuantity {
-    fn equal_or_greater_value(&self, other: &EtherQuantity) -> bool {
-        self >= other
+    fn compare_to(&self, other: &EtherQuantity) -> i8 {
+        if self < other {
+            return -1;
+        }
+        if self > other {
+            return 1;
+        }
+        return 0;
     }
 }
 impl Asset for Erc20Token {
-    fn equal_or_greater_value(&self, other: &Erc20Token) -> bool {
-        self.token_contract == other.token_contract && self.quantity >= other.quantity
```

Comment:
Note: this test should not be required as our query on btsieve only returns the correct transaction for the correct Erc20Token: 

https://github.com/comit-network/comit-rs/blob/25aa922fbd668030b617f79ac0a0f60645dcf0d6/cnd/src/swap_protocols/rfc003/ethereum/htlc_events.rs#L176-L216